AutocompleteClient/FW/Logic/ClientID/ClientIDGenerator.swift

```diff
@@ -0,0 +1,16 @@
+//
+//  ClientIDGenerator.swift
+//  AutocompleteClient
+//
+//  Copyright © Constructor.io. All rights reserved.
+//  http://constructor.io/
+//
+
+import Foundation
+import UIKit
+
+class ClientIDGenerator: UserIDGenerator{
+    func generateClientID() -> String?{
+        return UIDevice.current.identifierForVendor?.uuidString
```

Comment:
This `UIDevice.current.identifierForVendor` according to Apple is the same across all apps.  

For constructor.io's purposes we need the client identifier to be different in different apps.  This is a security consideration for our customers.  We do not want to be able to track user behavior across different customers.

For every `device/app` combination that uses constructor, we need a different client ID.  We are also fine if the client ID changes with every install.  In the android client and in the JS client we use random UUID's and store them